.github/workflows/publish.yml

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This action will build and publish branch to
+# https://shiro.apache.org/
+name: Stage current branch
+
+on:
+  workflow_dispatch: {}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 'Checkout main'
+        uses: actions/checkout@v4
+        with:
+          path: 'src'
+          persist-credentials: false
+
+      - uses: actions/setup-java@v4
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+
+      - run: mvn process-classes --show-version --errors --batch-mode --no-transfer-progress
+        working-directory: ./src
+
+      - name: 'Checkout target'
+        uses: actions/checkout@v4
+        with:
+          repository: 'apache/shiro-site'
+          ref: 'asf-site'
+          path: 'asf-site'
+
+      - name: 'Copy build to site repo'
+        run: |
+          rsync -a src/target/website/. ./asf-site
```

Comment:
What about deleting files?

Reply:
I think we should append ```
--delete
```

Reply:
I'm not sure, that _could_ cause 404's, but that might be better than orphaned files?

Reply:
I think Ben is correct. The automated flow needs to be the same as the [manual one](https://github.com/apache/shiro-site/blob/main/README.adoc) and needs `-avP --delete --exclude .git` cmd line arguments

Reply:
Ahh! It's in the doc, so I'll update the script.

Longer term we may want to think about how we prevent 404's but... it hasn't been an issue so I don't think we need to worry about it right now :D